pandas/tests/frame/test_convert_to.py

```diff
@@ -186,6 +186,18 @@ def test_to_records_with_categorical(self):
                                 dtype=[('index', '=i8'), ('0', 'O')])
         tm.assert_almost_equal(result, expected)
 
+    def test_to_records_with_strings_as_bytes(self):
+
+        # GH18146
+
+        df = DataFrame({'A': [1, 2], 'B': ['abc', 'defg']},
```

Comment:
can you test with some mixed object dtypes that include strings, e.g. ``[1, '1']``, this should preserve object dtype.

Reply:
maybe just add another column to your test frame